lib/contracts/avs/deploy.go

```diff
@@ -0,0 +1,255 @@
+func Deploy(ctx context.Context, backend *ethbackend.Backend) (common.Address, *ethtypes.Receipt, error) {
+	chainID, err := backend.ChainID(ctx)
+	if err != nil {
+		return common.Address{}, nil, errors.Wrap(err, "chain id")
+	}
+
+	cfg, err := getDeployCfg(chainID.Uint64())
+	if err != nil {
+		return common.Address{}, nil, errors.Wrap(err, "get deployment config")
+	}
+
+	return deploy(ctx, cfg, backend)
+}
+
+// DeployDevnet deploys the devnet AVS contract and returns the address receipt.
+func DeployDevnet(ctx context.Context, backend *ethbackend.Backend) (common.Address, *ethtypes.Receipt, error) {
+	chainID, err := backend.ChainID(ctx)
+	if err != nil {
+		return common.Address{}, nil, errors.Wrap(err, "chain id")
+	}
+
+	if chainids.IsMainnetOrTestnet(chainID.Uint64()) {
+		return common.Address{}, nil, errors.New("not a devnet")
+	}
+
+	return deploy(ctx, devnetDeployCfg(), backend)
+}
+
+// Deploy deploys the AVS contract and returns the deployed contract's address and the transaction receipt.
+func deploy(ctx context.Context, cfg DeploymentConfig, backend *ethbackend.Backend) (common.Address, *ethtypes.Receipt, error) {
+	if err := cfg.Validate(); err != nil {
+		return common.Address{}, nil, errors.Wrap(err, "validate config")
+	}
+
+	deployerTxOpts, err := backend.BindOpts(ctx, cfg.Deployer)
+	if err != nil {
+		return common.Address{}, nil, errors.Wrap(err, "bind deployer opts")
+	}
+
+	ownerTxOpts, err := backend.BindOpts(ctx, cfg.Owner)
+	if err != nil {
+		return common.Address{}, nil, errors.Wrap(err, "bind owner opts")
+	}
+
+	enc, err := packInitialzer(cfg)
+	if err != nil {
+		return common.Address{}, nil, err
+	}
+
+	impl, tx, _, err := bindings.DeployOmniAVS(deployerTxOpts, backend, cfg.Eigen.DelegationManager, cfg.Eigen.AVSDirectory)
+	if err != nil {
+		return common.Address{}, nil, errors.Wrap(err, "deploy avs impl")
+	}
+	_, err = backend.WaitMined(ctx, tx)
+	if err != nil {
+		return common.Address{}, nil, errors.Wrap(err, "wait mined avs proxy")
+	}
+
+	proxy, tx, _, err := bindings.DeployTransparentUpgradeableProxy(deployerTxOpts, backend, impl, cfg.ProxyAdmin, enc)
+	if err != nil {
+		return common.Address{}, nil, errors.Wrap(err, "deploy avs proxy")
+	}
+	receipt, err := backend.WaitMined(ctx, tx)
+	if err != nil {
+		return common.Address{}, nil, errors.Wrap(err, "wait mined avs proxy")
+	}
+
+	avs, err := bindings.NewOmniAVS(proxy, backend)
+	if err != nil {
+		return common.Address{}, nil, errors.Wrap(err, "bind avs")
+	}
+
+	if !cfg.AllowlistEnabled {
+		tx, err = avs.DisableAllowlist(ownerTxOpts)
+		if err != nil {
+			return common.Address{}, nil, errors.Wrap(err, "disable allowlist")
+		}
+
+		_, err = backend.WaitMined(ctx, tx)
+		if err != nil {
+			return common.Address{}, nil, errors.Wrap(err, "wait mined disable allowlist")
+		}
+	}
+
+	tx, err = avs.SetMetadataURI(ownerTxOpts, cfg.MetadataURI)
+	if err != nil {
+		return common.Address{}, nil, errors.Wrap(err, "set metadata uri")
+	}
```

Comment:
Might be worth adding metadata uri and and allowlist enabled / disabled to initializer, so we don't need admin txs here. 

And we might have admin controls runnable separately, so we can use this flow (with fireblocks keys) to call admin functions